File: Modules/Notifications/Http/Resources/NotificationResource.php
<?php

namespace Modules\Notifications\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;

class NotificationResource extends JsonResource
{
    public function toArray($request): array
    {
        return [
            'id' => $this->id,
            'title' => $this->data['title'] ?? 'Notification',
            'body' => $this->data['body'] ?? '',
            'type' => $this->data['type'] ?? 'info',
            'icon' => $this->data['icon'] ?? 'bell',
            'read_at' => $this->read_at,
            'created_at' => $this->created_at,
            'created_at_human' => $this->created_at->diffForHumans(),
        ];
    }
}

File: Solid.Api/Features/Settings/NotificationResource.cs
using Solid.Api.Common;
using Solid.Api.Database.Entities;

namespace Solid.Api.Features.Settings;

public static class NotificationResource
{
    public static object From(Notification notification)
    {
        var data = JsonPayload.ParseObject(notification.Data);

        return new
        {
            id = notification.Id,
            title = data.GetValueOrDefault("title") ?? "Notification",
            body = data.GetValueOrDefault("body") ?? "",
            type = data.GetValueOrDefault("type") ?? "info",
            icon = data.GetValueOrDefault("icon") ?? "bell",
            read_at = notification.ReadAt,
            created_at = notification.CreatedAt,
            created_at_human = ""
        };
    }
}
